packages/moonstone/internal/Picker/Picker.js

```diff
@@ -685,9 +717,11 @@ const PickerBase = class extends React.Component {
 	}
 
 	calcJoinedLabel (valueText) {
-		const {orientation} = this.props;
-		const hint = orientation === 'horizontal' ? $L('change a value with left right button') : $L('change a value with up down button');
-		return `${valueText} ${hint}`;
+		const
+			{['aria-label']: ariaLabel, orientation} = this.props,
+			hint = orientation === 'horizontal' ? $L('change a value with left right button') : $L('change a value with up down button');
+
+		return `${valueText} ${(ariaLabel == null) ? hint : ariaLabel}`;
```

Comment:
There's a small, hidden bug in here we should address as well. In `render()`, if the picker isn't joined `aria-label` will always be set to `null`. To avoid this, we should always call `calcJoinedLabel` (which should probably be renamed to `calcAriaLabel`) and when `joined`, run this logic and when `!joined` simply return `aria-label`.

Reply:
Also, I think that if `aria-label` is set, we should use it "as is" without prepending the value. What do you think?

The resulting logic (with the above comment) would be:

```
if (!joined || ariaLabel != null) {
  return ariaLabel;
}

// existing valuText + hint logic
```

Reply:
@ryanjduffy 
That's a good point.
I changed the name from calcJoinedLabel to calcAriaLabel, and I put some of your code.
I have one question.

if (!joined || ariaLabel != null) {
  return ariaLabel;
}

In the above code, ariaLabel is unconditionally set if aria-label is not null (TimePicker, DatePicker). I don't think this is expected.

What do you think of the code below?

if (!joined && !(ariaLabel == null)) {
  return ariaLabel;
}